### ISSUE TITLE
Type narrowing not working with properties

### ISSUE THREAD

**Issue by LostLuma** [microsoft/pyright#3404]
**Describe the bug**

Using a property that has a literal return type does not narrow types.

**To Reproduce**

Create two or more classes with a property that returns a unique literal value each.

**Expected behavior**

Type narrowing should narrow which type we're dealing with instead of ignoring the narrow attempt.

**Screenshots or Code**

Broken example:
```py
class A:
    @property
    def type(self) -> Literal[0]:
        return 0


class B:
    @property
    def type(self) -> Literal[1]:
        return 1


def test(x: A | B) -> None:
    if x.type == 1:
        reveal_type(x)  # Type of "x" is "A | B" Pyright
```

Working example:
```py
class A:
    type: Literal[0] = 0


class B:
    type: Literal[1] = 1


def test(x: A | B) -> None:
    if x.type == 1:
        reveal_type(x)  #Type of "x" is "B" Pyright
```

**VS Code extension or command-line**
Pyright v1.1.241 using VS Code

**Comment by erictraut** [comment]
This will be supported in the next release of pyright.

**Comment by LostLuma** [comment]
Thank you!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.242, which I just published. It will also be included in the next release of pylance.
